crates/oxc_minifier/tests/oxc/folding.rs

```diff
@@ -8,6 +8,12 @@ fn addition_folding() {
     test("x+''", "x+'';");
 }
 
+#[test]
+fn typeof_folding() {
+    test("typeof x === 'undefined'", "void 0===x;");
```

Comment:
This is dangerous. People use `typeof x` to determine if `x` exists.
For examples:

```JavaScript
if(typeof process === 'undefined') {
// not in node.js
}
```

If the definition of `x` is found in the context, then such an operation is feasible.

```JavaScript
function foo(x) {
    typeof x === 'undefined'; // We know that the variable x definitely exists.
}
```